### PR TITLE
[#170] Rename FlatSlotId and EpochIndex

### PR DESCRIPTION
**Pull request by ruhatch**
This is the first PR of #170 that does a couple of simple renamings and should be fairly uncontroversial

### CODE REVIEW

File: cardano-ledger/cardano-ledger.cabal [hunk 1]
```diff
@@ -84,7 +84,7 @@ library
                        Cardano.Chain.Genesis.NonAvvmBalances
                        Cardano.Chain.Genesis.Spec
 
-                       Cardano.Chain.Slotting.EpochIndex
+                       Cardano.Chain.Slotting.EpochNo
```
Comment (dnadales) [inline cardano-ledger/cardano-ledger.cabal]: Probably a bit late, and I think there are other packages that also use `EpochNo` and `SlotNo`, but to me these read as "epoch no" and "slot no" :) I just wonder why didn't we choose something like `EpochNumber` or `EpochNr` :man_shrugging:

**Review by dnadales — APPROVED** [review]
:wave: `FlatSlotId` :tada:

LFUTM (looks fairly uncontroversial to me) ;)

File: cardano-ledger/cardano-ledger.cabal (hunk 1 above)
Comment (ruhatch) [reply cardano-ledger/cardano-ledger.cabal]: No, you're right, and [here's](https://chrisdone.com/posts/german-naming-convention/) the blog post to prove it. I will update this :)

**Pull request merged**